Sofa/Component/SolidMechanics/FEM/Elastic/src/sofa/component/solidmechanics/fem/elastic/TetrahedronFEMForceField.inl

```diff
@@ -1832,56 +1834,78 @@ void TetrahedronFEMForceField<DataTypes>::drawTrianglesFromRangeOfTetrahedra(
             p[vId] = x[(*it)[vId]];
         }
 
-        const Coord center = (p[0] + p[1] + p[2] + p[3]) * 0.125;
-
-        if ( !showWireFrame )
+        if ( !showWireFrame && d_showElementGapScale.getValue() != 0.0 )
```

Comment:
d_showElementGraph.getValue() imply a complete data dependency graph tracking. It shouldn't be used in a loop unless the value can change between every loop iterations. To improve, do bool showElementsGapScale =d_showElementsGapScale.getValue() out of the loop.